cloudinit/config/cc_apt_configure.py

```diff
@@ -499,12 +501,19 @@ def add_apt_key_raw(key, file_name, hardened=False, target=None):
         raise
 
 
-def add_apt_key(ent, target=None, hardened=False, file_name=None):
+@functools.lru_cache(maxsize=1)
+def _ensure_gpg(cloud):
+    if not shutil.which("gpg"):
```

Comment:
what's the difference between this, and subp.which?

Reply:
`shutil` is the standard library version. They both do basically the same thing. `subp` version has some slightly different logic that isn't relevant in most cases. I think we can probably move away from the `subp` version at some point.

Reply:
I thought the main reason were using our own wrapper is ease of testing

Reply:
How do see that easing testing?